tests/test_builders/test_build_warnings.py

```diff
@@ -70,3 +70,19 @@ def test_texinfo_warnings(app, warning):
     app.build(force_all=True)
     warnings_exp = TEXINFO_WARNINGS.format(root=re.escape(app.srcdir.as_posix()))
     _check_warnings(warnings_exp, warning.getvalue())
+
+
+def test_uncacheable_config_warning(make_app, tmp_path):
+    """Test than an unpickleable config value raises a warning."""
+    tmp_path.joinpath('conf.py').write_text("""
+my_config = lambda: None
+show_warning_types = True
+def setup(app):
+    app.add_config_value('my_config', None, 'env')
+    """, encoding='utf-8')
+    tmp_path.joinpath('index.rst').write_text('Test\n====\n', encoding='utf-8')
+    app = make_app(srcdir=tmp_path)
+    app.build()
+    assert strip_colors(app.warning.getvalue()).strip() == (
```

Comment:
This kind of construction definitely needs a helper function or attribute on SphinxTestApp where you automatically get the warning strings without colors etc.

I'm thinking of something that could be used as the LineMatcher of pytester (that would also help if we could match messages easily using regexes or using `<=` for blocks of messages, I think).

Reply:
I would certainly welcome this, in sphinx core and for extension development 😄

Reply:
I can do it after I've launched the simulations that I need to launch for today.

Reply:
> I'm thinking of something that could be used as the LineMatcher of pytester

Although ideally I would still like the option to "dumbly" assert against the exact list of warnings (the only annoyance being the` srcdir` prefix on warnings)